sinks/influxdb/driver.go

```diff
@@ -303,12 +303,12 @@ func newClient(c config) (influxdbClient, error) {
 func new(c config) (sink_api.ExternalSink, error) {
 	client, err := newClient(c)
 	if err != nil {
-		fmt.Errorf("issues while creating an InfluxDB sink: %v, will retry on use", err)
+		err = fmt.Errorf("issues while creating an InfluxDB sink: %v, will retry on use", err)
```

Comment:
Probably?  The comment below says it is ok for the client to be nil, but I think you are right and it should return a nil sink in case of error.